java_io_block/main_task/src/main/java/MyFileReader.java

```diff
@@ -0,0 +1,101 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class MyFileReader {
+
+    /**
+     * Try to read file and write all file strings into the list
+     * @param path - absolute path to the txt file
+     * @return list of file strings
+     */
+    public static List<String> readFile(String path) {
+        try {
+            BufferedReader br = Files.newBufferedReader(Paths.get(path));
+            return br.lines().collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new IllegalStateException("Cannot read file", e);
+        }
+    }
+
+    public static int countDirectories(List<String> listOfLines) {
+        int numOfDirs = 0;
+        for (String line : listOfLines) {
+            if (line.contains(DirectoryReader.directoryPrefix)) {
+                numOfDirs++;
+            }
+        }
+        return numOfDirs;
+    }
+
+    public static int countFiles(List<String> listOfLines) {
+        int numOfFiles = 0;
+        for (String line : listOfLines) {
+            if (line.contains(DirectoryReader.filePrefix) && !line.contains(DirectoryReader.directoryPrefix)) {
+                numOfFiles++;
+            }
+        }
+        return numOfFiles;
+    }
+
+    public static double averageFilesAmountInDir(List<String> listOfLines) {
+        return (double) countFiles(listOfLines) / countDirectories(listOfLines);
+    }
+
+    public static double averageFileNameSize(List<String> listOfLines) {
+        int numOfLines = 0;
+        int numOfElems = 0;
+        for (String line : listOfLines) {
+            if (line.contains(DirectoryReader.filePrefix) && !line.contains(DirectoryReader.directoryPrefix)) {
+                Pattern pattern = Pattern.compile("--(.+?)(\\.[^.]*$|$)");// find only file's name without extension
+                Matcher matcher = pattern.matcher(line);
+                if (matcher.find()) {
```

Comment:
what will happen if there will be no lines which match pattern

what `(double) 0/0` will return and how you would handle it?